azurerm/internal/services/mssql/resource_arm_mssql_virtual_machine.go

```diff
@@ -0,0 +1,425 @@
+				},
+			},
+		},
+		Tags: tags.Expand(d.Get("tags").(map[string]interface{})),
+	}
+
+	if sqlVirtualMachineGroupResourceID, ok := d.GetOk("sql_virtual_machine_group_id"); ok {
+		parameters.Properties.SQLVirtualMachineGroupResourceID = utils.String(sqlVirtualMachineGroupResourceID.(string))
+	}
+
+	future, err := client.CreateOrUpdate(ctx, id.ResourceGroup, id.Name, parameters)
+	if err != nil {
+		return fmt.Errorf("Failure in creating Sql Virtual Machine (Sql Virtual Machine Name %q / Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
+	}
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Failure in waiting for creation of Sql Virtual Machine (Sql Virtual Machine Name %q / Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.Name, "*")
+	if err != nil {
+		return fmt.Errorf("Failure in retrieving Sql Virtual Machine (Sql Virtual Machine Name %q / Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
+	}
+	if resp.ID == nil {
+		return fmt.Errorf("Cannot read Sql Virtual Machine (Sql Virtual Machine Name %q / Resource Group %q) ID", id.Name, id.ResourceGroup)
+	}
+	d.SetId(*resp.ID)
+
+	return resourceArmMsSqlVirtualMachineRead(d, meta)
+}
+
+func resourceArmMsSqlVirtualMachineRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).MSSQL.SQLVirtualMachinesClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.MssqlVmID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.Name, "*")
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] Sql Virtual Machine %q does not exist - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Failure in reading Sql Virtual Machine (Sql Virtual Machine Name %q / Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
+	}
+
+	if props := resp.Properties; props != nil {
+		d.Set("virtual_machine_id", props.VirtualMachineResourceID)
+		d.Set("sql_license_type", string(props.SQLServerLicenseType))
+		if err := d.Set("auto_patching", flattenArmSqlVirtualMachineAutoPatching(props.AutoPatchingSettings)); err != nil {
+			return fmt.Errorf("Failure in setting `auto_patching`: %+v", err)
+		}
+		if err := d.Set("key_vault_credential", flattenArmSqlVirtualMachineKeyVaultCredential(props.KeyVaultCredentialSettings, d)); err != nil {
+			return fmt.Errorf("Failure in setting `key_vault_credential`: %+v", err)
+		}
+
+		d.Set("r_services_enabled", props.ServerConfigurationsManagementSettings.AdditionalFeaturesServerConfigurations.IsRServicesEnabled)
+		d.Set("sql_connectivity_port", props.ServerConfigurationsManagementSettings.SQLConnectivityUpdateSettings.Port)
+		d.Set("sql_connectivity_type", props.ServerConfigurationsManagementSettings.SQLConnectivityUpdateSettings.ConnectivityType)
```

Comment:
could we nil check here
```suggestion
		if mgmtSettings := props.ServerConfigurationsManagementSettings; mgmtSettings != nil {
			if cfgs := mgmtSettings.AdditionalFeaturesServerConfigurations; cfgs != nil {
				d.Set("r_services_enabled", mgmtSettings.AdditionalFeaturesServerConfigurations.IsRServicesEnabled)
			}
		d.Set("sql_connectivity_port", mgmtSettings.SQLConnectivityUpdateSettings.Port)
		d.Set("sql_connectivity_type", mgmtSettings.SQLConnectivityUpdateSettings.ConnectivityType)
```